src/libstd/sys/unix/condvar.rs

```diff
@@ -116,7 +173,7 @@ impl Condvar {
                    (sys_now.tv_usec * 1000) as libc::c_long;
         let extra = (nsec / 1_000_000_000) as libc::time_t;
         let nsec = nsec % 1_000_000_000;
-        let seconds = dur.as_secs() as libc::time_t;
+        let seconds = wrapping_cast_to_time_t(dur.as_secs());
```

Comment:
Yes, if `time_t` is more than 32 bit (as in all modern OSes), this cast is not needed.

However, I'd keep this cast, because
* it is not performance critical
* makes code easier to read (reader does not need to check that there's no overflow)
* makes OSX/Linux versions more like to each other